### PR TITLE
Add a `loadComponent()` helper which provides a lot of the boilerplating

### PR DESCRIPTION
**Pull request by flamerohr**
Reduces the code normally done when needing a component from Injector

### CODE REVIEW

File: client/src/lib/dependency-injection/tests/loadComponent-test.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    expect(testComponent.textContent).toBe('abc');
+  });
+
+  it('should return a rendered TestComponent when loading TestComponent from Injector', () => {
+    const Temp = loadComponent('TestComponent', null);
+    const temp = ReactTestUtils.renderIntoDocument(<Temp />);
+    const testComponent = ReactTestUtils.findRenderedDOMComponentWithTag(temp, 'span');
+
+    expect(testComponent.textContent).toBe('def');
+  });
```
Comment (unclecheese) [inline client/src/lib/dependency-injection/tests/loadComponent-test.js]: Maybe add a test for a non-falsey `context` param?

**Comment by unclecheese** [comment]
Looks like the tests are having a hard time with the late `require()` of React.

File: client/src/lib/dependency-injection/tests/loadComponent-test.js (hunk 1 above)
Comment (flamerohr) [reply client/src/lib/dependency-injection/tests/loadComponent-test.js]: that's sort of what's happening with passing `undefined` :)
> Default function parameters allow formal parameters to be initialized with default values if no value or undefined is passed.
Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters

But fair point, that is a bit magical to read - I'll update it to be explicit

**Comment by flamerohr** [comment]
oh, case-sensitivity issue :( uppercase `R` broke that on travis

**Comment by flamerohr** [comment]
Screenshot of the `NotFoundComponent` when it's shown
screen shot 2017-12-06 at 12 42 09 pm

**Comment by unclecheese** [comment]
Thank you! I was going to ask for that :)

**Pull request merged**